Assets/Tests/Runtime/core/SingletonTests.cs

```diff
@@ -0,0 +1,49 @@
+﻿using UnityEngine;
+using UnityEngine.TestTools;
+using NUnit.Framework;
+using System.Collections;
+
+public class SingletonTests {
```

Comment:
This is way over commented to the point where it is hard to read the code. In the future try to keep comments minimal.

Reply:
The comments document subtleties not apparent from reading the code. So I think it's best to keep their content, but I'll try to make them more concise.